### PR TITLE
Support X509 Federation

### PR DESCRIPTION
**Pull request by EZIOJQ**
This PR adds support for X.509 federation support on workload identity pool. It fixes https://github.com/hashicorp/terraform-provider-google/issues/18812. 

For more details about how this feature works, check out the public doc here https://cloud.google.com/iam/docs/workload-identity-federation-with-x509-certificates

```release-note:enhancement
iambeta: added `x509` field to `google_iam_workload_identity_pool_provider ` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb [hunk 1]
```diff
@@ -240,4 +240,96 @@ resource "google_iam_workload_identity_pool_provider" "my_provider" {
...
...NVBAMMGkdvb2dsZSAqKlRlc3RpbmcqKiBSb290IENBMCAXDTIwMDEwMTAw\nMDAwMFoYDzIxMzAwMTAxMDAwMDAwWjBcMRcwFQYDVQQKEw5Hb29nbGUgVEVTVElO\nRzEcMBoGA1UECxMTR29vZ2xlIFRFU1RJTkcgdW5pdDEjMCEGA1UEAwwaR29vZ2xl\nICoqVGVzdGluZyoqIFJvb3QgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK\nAoIBAQDHqdf17AA4iwl+HGtTD/qJJv9XZTjzjHiaRYWuGQ3iddf1LhYklLqHvpqO\nCBlhkg6NIZmoDPKicEi8pfGSzp6btcElyrr1ekECk5jEBcdl6tX/gTSYv7v1h9Dk\nSJDHBAnoSsVW0/PNwI+YGEE2kizGMeg1moXlTHEB+yeGCik/+4eVRas/+wrlrTE5\nlMFMq8WhdnBx6udcc/BEauvlTybHaN3rJUVpVrJWeVoPGGtXR6MJrdbVScn9GYOL\nP6sXMPTr+pTY6ebrjs7K/wTVQqLJ1zArCvAEH8FvuAhiI19yM1uBRAds/fJbSUvF\nsEiIlDC4y7DAK7yWHUAjQIBV1xhDAgMBAAGjejB4MA4GA1UdDwEB/wQEAwICBDAd\nBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDwYDVR0TAQH/BAUwAwEB/zAZ\nBgNVHQ4EEgQQxEHfoOjPZleWgcpHJnyE3TAbBgNVHSMEFDASgBDEQd+g6M9mV5aB\nykcmfITdMA0GCSqGSIb3DQEBCwUAA4IBAQBcECwxgaK4ZgYO7guayK5QRpTb3Y6z\npkmSOkj9h1+HF5Ch/o5FiweJi8k28h9Mfz//gKU2cXXWfXuY81CFEBstjw7jYt3+\nd4owS5sYKu5WswGj4jsQXRrt0tO0+0UngP560eggFCyLChG75lp54r92hTPk4fY4\nvarURZH7X0jg9W7MO6E6/HmKdMIxanuWdkbpPe6kj5I7SNvVhqsncoga8iZJ6QK/\nrsC2LTax4dxUUSkP5vmwwiYbgXYbk9JcXI+OyVcMvVxN/akI9/JgYHOol6NdChTB\nwU0yjNb9B5TrgtP1/fs+LugINrN1R1hgVHDVlE4mwHej+5XL6m9xAxkc\n-----END CERTIFICATE-----"
```
Comment (melinath) [inline mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb:295]: Why not use files here as well?

**Review by melinath — COMMENTED** [review]
@EZIOJQ This looks good overall. Just wanted to double-check that this field is GA and doesn't require an allowlist? The linked docs say it's in preview and that you need to contact folks to get it enabled - if that's still the case, we shouldn't add it to the provider yet.

File: mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb (hunk 1 above)
Comment (EZIOJQ) [reply mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb]: Updated this as well

**Comment by EZIOJQ** [comment]
> @EZIOJQ This looks good overall. Just wanted to double-check that this field is GA and doesn't require an allowlist? The linked docs say it's in preview and that you need to contact folks to get it enabled - if that's still the case, we shouldn't add it to the provider yet.

This feature will be GA soon, and we will remove the allowlist in next few weeks. We can merge the change after the allowlist is removed

File: mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb [hunk 2]
```diff
@@ -240,4 +240,96 @@ resource "google_iam_workload_identity_pool_provider" "my_provider" {
...
+  attribute_mapping                  = {
+    "google.subject"        = "assertion.subject.dn.cn"
+  }
+  x509 {
+    trust_store {
+        trust_anchors {
+            pem_certificate = file("test-fixtures/trust_anchor.pem")
+        }
+        trust_anchors {
+            pem_certificate = file("test-fixtures/intermediate_ca.pem")
```
Comment (melinath) [inline mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb:327]: I think this previously was updating the certs but that was lost in the transition to file-based?

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb (hunk 2 above)
Comment (EZIOJQ) [reply mmv1/third_party/terraform/services/iambeta/resource_iam_workload_identity_pool_provider_test.go.erb:327]: Good catch! Updated the test here

**Comment by EZIOJQ** [comment]
@melinath, we have this allowlist for half a year now.  Are we okay to merge the change given it's long-term allowlist?

**Review by melinath — COMMENTED** [review]
Since the allowlist is about to come off anyway, I'd rather just wait until it's removed.

**Comment by melinath** [comment]
after discussion, we've decided to move forward with the allowlist in place. Marking for review

**Review by melinath — APPROVED** [review]
Seems straightforward and the tests are passing.

**Pull request merged**